crates/viewer/re_component_ui/src/timeline_columns.rs

```diff
@@ -0,0 +1,126 @@
+use re_data_ui::item_ui::timeline_button;
+
+use re_log_types::TimelineName;
+use re_types::blueprint::components::TimelineColumn;
+use re_ui::{HasDesignTokens as _, UiExt as _};
+use re_viewer_context::{MaybeMutRef, ViewerContext};
+
+pub fn edit_or_view_columns_singleline(
+    ctx: &ViewerContext<'_>,
+    ui: &mut egui::Ui,
+    columns: &mut MaybeMutRef<'_, Vec<TimelineColumn>>,
+) -> egui::Response {
+    ui.horizontal(|ui| {
+        for col in columns.iter() {
+            if *col.visible {
+                timeline_button(ctx, ui, &TimelineName::new(&col.timeline));
+            }
+        }
+    })
+    .response
+}
+
+pub fn edit_or_view_columns_multiline(
```

Comment:
would be _nice_ to do some code sharing with text_log_columns, looks very similar from afar 🤔